zerocopy-derive/tests/enum_from_zeros.rs

```diff
@@ -44,9 +44,58 @@ enum ImplicitNonFirstVariantIsZero {
     B,
 }
 
+util_assert_impl_all!(ImplicitNonFirstVariantIsZero: imp::FromZeros);
+
 #[derive(imp::FromZeros)]
 #[repr(u64)]
 enum LargeDiscriminant {
     A = 0xFFFF_FFFF_FFFF_FFFF,
     B = 0x0000_0000_0000_0000,
 }
+
+util_assert_impl_all!(LargeDiscriminant: imp::FromZeros);
+
+#[derive(imp::FromZeros)]
+#[repr(C)]
+enum FirstVariantIsZeroable {
+    A(u32),
+    B { foo: u32 },
+}
+
+util_assert_impl_all!(FirstVariantIsZeroable: imp::FromZeros);
+
+#[derive(imp::FromZeros)]
+#[repr(u8)]
+enum FirstVariantIsZeroableSecondIsNot {
+    A(bool),
+    B(::core::num::NonZeroU8),
+}
+
+util_assert_impl_all!(FirstVariantIsZeroableSecondIsNot: imp::FromZeros);
+
+// MSRV does not support data-carrying enum variants with explicit discriminants
+#[cfg(not(__ZEROCOPY_TOOLCHAIN = "msrv"))]
+#[derive(imp::FromZeros)]
+#[repr(u8)]
+enum ImplicitFirstVariantIsZeroable {
+    A(bool),
+    B(::core::num::NonZeroU8) = 1,
+}
+
+// MSRV does not support data-carrying enum variants with explicit discriminants
+#[cfg(not(__ZEROCOPY_TOOLCHAIN = "msrv"))]
+util_assert_impl_all!(ImplicitFirstVariantIsZeroable: imp::FromZeros);
+
+// MSRV does not support data-carrying enum variants with explicit discriminants
+#[cfg(not(__ZEROCOPY_TOOLCHAIN = "msrv"))]
+#[derive(imp::FromZeros)]
+#[repr(i8)]
+enum ImplicitNonFirstVariantIsZeroable {
+    A(::core::num::NonZeroU8) = 1,
+    B = -1,
+    C(bool),
+}
+
+// MSRV does not support data-carrying enum variants with explicit discriminants
+#[cfg(not(__ZEROCOPY_TOOLCHAIN = "msrv"))]
+util_assert_impl_all!(ImplicitNonFirstVariantIsZeroable: imp::FromZeros);
```

Comment:
Maybe put these in a module so you can factor out the `#[cfg(not(__ZEROCOPY_TOOLCHAIN = "msrv"))]`?

Reply:
Done